client/src/partials/reports/global_transaction/global_transaction.js

```diff
@@ -133,45 +142,64 @@ angular.module('bhima.controllers')
     }
 
     function search () {
-      if (!$scope.model.account_id) { return; }
-      $scope.mode = $scope.model.account_id !== 0 ? 'selected' : 'all';
+      $scope.mode = ($scope.model.account_id && $scope.model.account_id > 0) ? 'selected' : 'all';
+      
+      if ($scope.model.account_id && $scope.model.account_id > 0) {
+        $scope.model.account_number = $scope.accounts.get($scope.model.account_id).account_number;       
+      } else {
+        $scope.model.account_number = 0;
+      }
+
       var qo = {
```

Comment:
Putting things in a JSON object like this is unnecessary, this will generate a URL in this format: `/reports/allTrans/?{key:"value",key2:"value2"}`.  In the server logs, this is difficult to read.  It will we better if you composed these as parameters: `/reports/allTrans?key=value&key2=value2`.